utils/oc.py

```
@@ -73,7 +73,8 @@ def __init__(self, server, token, jh=None, settings=None,
 
         self.oc_base_cmd = oc_base_cmd
         # calling get_version to check if cluster is reachable
-        self.get_version()
+        if server != 'server':
+            self.get_version()
```

This is way too fragile.

---

@BumbleFeng @apahim is not happy with this solution, but when I have a better one I will add comments.

---

I mean, that could have been any string... There's nothing special about `'server'`.

---

'server' comes from saasherder https://github.com/app-sre/qontract-reconcile/blob/master/utils/saasherder.py#L328
It only needs to run oc to verify the template file, doesn't need to actually run towards a real cluster.
We could either skip check  get_version for saasherder or catch and swallow the error when init oc in saasherder.

---

wdyt?
`
        if server.startswith('https://api'):
            self.get_version()
`

---

Let's take a step back here for a minute. Do we understand why we are setting the server to `'server' `? Is that the right thing to do?

---

@apahim  If it is `server` it isn't valid and we would want to skip it.  Is there a positive case where we would just return `server`?

---

No, I don't think so I am experimenting with a try block now.  I will try this and test `sql-query` and all `openshift-*` integrations.  Unless you think this is incorrect?